File: 71-simplify-path/71-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        curr, ans = "", ""
        st = []
        path += "/"
        
        for c in path: 
            if c == '/':
                if len(curr) == 0:
                    continue
                elif curr == "..":
                    if len(st) > 0:
                        st.pop()
                
                elif curr != ".":
                    st.append(curr)
                curr = ""
            
            else:
                curr += c
        
        
        while len(st) > 0:
            ans = "/" + st[-1] + ans
            st.pop()
        
        if len(ans) == 0: return "/"
        return ans

File: 71-simplify-path/71-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        return solve(path);
    }
private:
    string solve(string path) {
        string curr, ans;
        stack<string> st;
        path.push_back('/');
        
        for(char c : path) {
            if(c == '/') {
                if(curr.empty())    continue;
                else if(curr == "..") {
                    if(!st.empty())   st.pop(); 
                }
                else if(curr != ".")    st.push(curr); 
                curr = "";
            }
            else
                curr.push_back(c);
        }
        
        while(!st.empty()) {
            ans = "/" + st.top() + ans;
            st.pop();
        }
        if(ans.empty()) return "/";
        return ans;
    }
};